zephyr/ra/ra_cfg/fsp_cfg/bsp/ra2a1/bsp_clock_cfg.h

```diff
@@ -0,0 +1,123 @@
+/*
+* Copyright (c) 2020 - 2024 Renesas Electronics Corporation and/or its affiliates
+* Copyright (c) 2024 TOKITA Hiroshi
+*
+* SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include <zephyr/devicetree.h>
+#include <zephyr/dt-bindings/clock/ra_clock.h>
+
+#ifndef BSP_CLOCK_CFG_H_
+#define BSP_CLOCK_CFG_H_
+
+#define BSP_CFG_CLOCKS_SECURE   (0)
+#define BSP_CFG_CLOCKS_OVERRIDE (0)
+#define BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(node_id, prop, default_value)                            \
+	(COND_CODE_1(DT_NODE_HAS_STATUS(node_id, okay), (DT_PROP(node_id, prop)), (default_value)))
+
+#define BSP_CFG_XTAL_HZ BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(DT_NODELABEL(xtal), clock_frequency, 0)
+
+#if DT_PROP(DT_NODELABEL(hoco), clock_frequency) == 24000000
+#define BSP_CFG_HOCO_FREQUENCY 0 /* HOCO 24MHz */
+#elif DT_PROP(DT_NODELABEL(hoco), clock_frequency) == 32000000
+#define BSP_CFG_HOCO_FREQUENCY 2 /* HOCO 32MHz */
+#elif DT_PROP(DT_NODELABEL(hoco), clock_frequency) == 48000000
+#define BSP_CFG_HOCO_FREQUENCY 4 /* HOCO 48MHz */
+#elif DT_PROP(DT_NODELABEL(hoco), clock_frequency) == 64000000
+#define BSP_CFG_HOCO_FREQUENCY 5 /* HOCO 64MHz */
+#else
+#error "Invalid HOCO frequency, only can be set to 24MHz, 32MHz, 48MHz, 64MHz"
+#endif
+
+#define BSP_CFG_PLL_SOURCE                                                                         \
+	BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(DT_NODELABEL(pll), source, RA_PLL_SOURCE_DISABLE)
+#define BSP_CFG_PLL_DIV BSP_CLOCK_PROP_HAS_STATUS_OKAY_OR(DT_NODELABEL(pll), div, RA_PLL_DIV_1)
+#define BSP_CFG_PLL_MUL                                                                            \
+	DT_NODE_HAS_STATUS(DT_NODELABEL(pll), okay)                                                \
+	? BSP_CLOCKS_PLL_MUL(DT_PROP_BY_IDX(DT_NODELABEL(pll), mul, 0),                            \
+			     DT_PROP_BY_IDX(DT_NODELABEL(pll), mul, 1))                            \
+	: BSP_CLOCKS_PLL_MUL(0, 0)
```

Comment:
@soburi , Could you please help to check commit https://github.com/zephyrproject-rtos/hal_renesas/pull/24/commits/4c4b6c711d3e38f00ace4f84610b8a937bf2966f and modify follow it, reason is this missing parenthesis "()" wrap around the macro so it make the multiplication value calculation "BSP_CLOCKS_PLL_MUL" incorrect

Reply:
I modified it to follow 4c4b6c711d3e38f00ace4f84610b8a937bf2966f changes.